tests/unit/reco_utils/recommender/test_fbt.py

```diff
@@ -0,0 +1,199 @@
+"""Tests for FBT methods."""
+import pytest
+import pandas as pd
+from reco_utils.recommender.fbt.fbt import FBT
+from pandas._testing import assert_frame_equal
+import numpy as np
+
+
+@pytest.fixture
+def model_class():
+    return FBT(col_user="user_id",
+               col_item="item_id",
+               col_score="score",
+               num_recos=1)
+
+
+def test_init(model_class):
+    assert model_class.col_user == "user_id"
+    assert model_class.col_item == "item_id"
+    assert model_class.col_score == "score"
+    assert model_class.num_recos == 1
+
+
+@pytest.fixture
+def train_data(model_class):
+    data_dict = {
+            model_class.col_user: [1, 1, 1, 2, 2, 3],
+            model_class.col_item: [11, 22, 33, 11, 33, 22],
+            f'{model_class.col_item}_name': ['aa', 'bb', 'cc',
+                                             'aa', 'cc', 'bb']
+        }
+    return pd.DataFrame(data_dict)
+
+
+@pytest.fixture
+def expected_fbt_model(model_class):
+    model_df = pd.DataFrame({
+            model_class.col_item: [11, 11, 22, 22, 33, 33],
+            f'{model_class.col_item}_paired': [22, 33, 11, 33, 11, 22],
+            model_class.col_score: [1, 2, 1, 1, 2, 1]
+        })
+    return model_df
+
+
+def test_fbt_fit(model_class, train_data, expected_fbt_model):
+    """Test fit() method to work and fail as expected."""
+    # test output of fit function
+    model_class.fit(train_data)
+    computed_fbt_model = model_class._model_df.reset_index(drop=True)
+
+    assert type(computed_fbt_model) == pd.DataFrame
+
+    # test for expected output columns
+    expected_column_names = {
+        model_class.col_item,
+        f'{model_class.col_item}_paired',
+        model_class.col_score
+    }
+    actual_column_names = set(computed_fbt_model.columns)
+    assert expected_column_names == actual_column_names
+
+    # test the output dataframe on train_df
+    assert_frame_equal(expected_fbt_model, computed_fbt_model)
```

Comment:
this is correct, however we typically put this on a different way, we use fixtures for data, not for a model. So columns names, input training data and predicted data would be fixtures. The model won't. 
For example in this function we would typically do:
```
def test_fbt_fit( ...  ):
    """Test fit() method to work and fail as expected."""
model = FBT(...)
    # test output of fit function
    model_class.fit(train_data)
    computed_fbt_model = model_class._model_df.reset_index(drop=True)

    assert type(computed_fbt_model) == pd.DataFrame

    # test for expected output columns
    expected_column_names = {
        model_class.col_item,
        f'{model_class.col_item}_paired',
        model_class.col_score
    }
    actual_column_names = set(computed_fbt_model.columns)
    assert expected_column_names == actual_column_names

    # test the output dataframe on train_df
    assert_frame_equal(expected_fbt_model, computed_fbt_model)
```